go/vt/orchestrator/inst/tablet_dao.go

```diff
@@ -36,27 +37,72 @@ var TopoServ *topo.Server
 // ErrTabletAliasNil is a fixed error message.
 var ErrTabletAliasNil = errors.New("tablet alias is nil")
 
-// TabletSetMaster designates the tablet that owns an instance as the master.
-func TabletSetMaster(instanceKey InstanceKey) error {
+// SwitchMaster makes the new tablet the master and proactively performs
+// the necessary propagation to the old master. The propagation is best
+// effort. If it fails, the tablet's shard sync will eventually converge.
+// The proactive propagation allows a competing Orchestrator from discovering
+// the successful action of a previous one, which reduces churn.
+func SwitchMaster(newMasterKey, oldMasterKey InstanceKey) error {
+	newMasterTablet, err := ChangeTabletType(newMasterKey, topodatapb.TabletType_MASTER)
+	if err != nil {
+		return err
+	}
+	// The following operations are best effort.
+	if newMasterTablet.Type != topodatapb.TabletType_MASTER {
+		log.Errorf("Unexpected: tablet type did not change to master: %v", newMasterTablet.Type)
+		return nil
+	}
+	_, err = TopoServ.UpdateShardFields(context.TODO(), newMasterTablet.Keyspace, newMasterTablet.Shard, func(si *topo.ShardInfo) error {
+		if proto.Equal(si.MasterAlias, newMasterTablet.Alias) && proto.Equal(si.MasterTermStartTime, newMasterTablet.MasterTermStartTime) {
+			return topo.NewError(topo.NoUpdateNeeded, "")
+		}
+
+		// We just successfully reparented. We should check timestamps, but always overwrite.
+		lastTerm := si.GetMasterTermStartTime()
+		newTerm := logutil.ProtoToTime(newMasterTablet.MasterTermStartTime)
+		if !newTerm.After(lastTerm) {
+			log.Errorf("Possible clock skew. New master start time is before previous one: %v vs %v", newTerm, lastTerm)
+		}
+
+		aliasStr := topoproto.TabletAliasString(newMasterTablet.Alias)
+		log.Infof("Updating shard record: master_alias=%v, master_term_start_time=%v", aliasStr, newTerm)
+		si.MasterAlias = newMasterTablet.Alias
+		si.MasterTermStartTime = newMasterTablet.MasterTermStartTime
+		return nil
+	})
+	// Don't proceed if shard record could not be updated.
```

Comment:
should this comment be removed, seeing that in the following code we return `nil` even on error?

Reply:
The comment applies to the error check and nil return. It says that we should not ignore this error. LMK if there is a better way to phrase it.